### ISSUE TITLE
Chain inference and evaluation to training

### ISSUE THREAD

**Issue by tjhunter** [ecmwf/WeatherGenerator#1925]
### Describe the task. Describe the task. It can be a feature, a set of experiments, documentation, etc.

Precious time is lost between the end of a training job and the start of inference and then again evaluation, because each of these require manual steps. As a result, a full cycle that could be done in 24 hours now takes usually 2-3 business days. 

The goal of this task: modify launch-slurm so that it can schedule training, inference and eval

General guidelines:
- it should not change the existing workflows (the current commands should still work, it is just adding new options)
- no resilience to failures
- it should not attempt to unify the configuration between the various steps. It should not require changes to the existing config system in terms of layout or behavior of options
- it should rely on slurm only (no ecflow, airflow, homegrown scheduler etc.)

The goal is to show at least a proof of concept on santis and juwels. Once this is debugged, we can add the other HPCs.

It should likely required a different slurm script for evaluation because the compute needs are very different.

### Area

- [ ] datasets, data readers, data preparation and transfer
- [ ] model
- [ ] science
- [x] infrastructure and engineering
- [x] evaluation, export and visualization
- [ ] documentation

**Comment by iluise** [comment]
This is a first step in this direction: https://github.com/ecmwf/WeatherGenerator/pull/1747

Now the eval parameters are hardcoded in the script, but we probably need to add the eval config parameters to the "main" config so people can choose what to run in advance. 

NB. The comparison _across_ run_ids will still be separate, but if the scores are already computed and stored in json that will take no time and the evaluation will be definitely much faster.

**Comment by clessig** [comment]
If we have a fine-tuning stage (which we currently have for all major lines of work), then this requires an overwrite config. Specifying this as well in the base-config for the pre-training seem complex (although not impossible if we agree that we have zero-level keys `pretraining`, `finetuning`, `evalution`; what when we have multiple fine-tuning stages?)

**Comment by tjhunter** [comment]
Do people have strong feelings if we start with [`training`|`pretraining`|`finetuning`] -> `inference` -> `evaluation`? 

To @clessig  's point:
- we do not have yet a well-defined full cycle in the config for training/pretraining/finetuning
- can someone advocate the case of automatically chaining `training` -> `finetuning` _right now_? I have heard clear pain points for inference and evaluation, but not so much for before yet.

**Comment by clessig** [comment]
> * d full cycle in the config for training/pretraining/finetuning

90% of our *current* training runs right now do pre-training + fine-tuning (since we do it for forecasting and SSL-type pre-training). It's unnecessary human effort and a source of error.

**Comment by sophie-xhonneux** [comment]
Can this please be raised in priority to P0!

**Comment by simone99n** [comment]
assign me

**Comment by tjhunter** [comment]
Implemented by Simone in https://gitlab.jsc.fz-juelich.de/esde/WeatherGenerator-private/-/commit/ef08ff563206526f0d4649c3167fd4b0e840b15e
